packages/diracx-web-components/src/components/shared/SearchBar/Utils.tsx

```diff
@@ -0,0 +1,356 @@
+import { Button, ButtonGroup, Box } from "@mui/material";
+
+import type {
+  SearchBarTokenEquation,
+  SearchBarToken,
+  InternalFilter,
+  SearchBarSuggestions,
+} from "../../../types";
+
+import type { EquationAndTokenIndex } from "./Types";
+
+/**
+ * @param tokenEquations The list of token equations to be verified.
+ * @param setTokenEquations A function to update the state of token equations.
+ * This function verifies the validity of each equation and updates their status.
+ */
+export function handleEquationsVerification(
+  tokenEquations: SearchBarTokenEquation[],
+  setTokenEquations: React.Dispatch<
+    React.SetStateAction<SearchBarTokenEquation[]>
+  >,
+) {
+  tokenEquations = tokenEquations.map(handleEquationVerification);
+
+  if (
+    tokenEquations.length > 0 &&
+    tokenEquations[tokenEquations.length - 1].status === "invalid" &&
+    tokenEquations[tokenEquations.length - 1].items.length < 3
+  ) {
+    tokenEquations[tokenEquations.length - 1].status = "waiting";
+  }
+
+  setTokenEquations([...tokenEquations]);
+}
+
+/**
+ * @param tokenEquation The equation to be verified.
+ * @returns The equation with its status updated based on its validity.
+ */
+function handleEquationVerification(
+  tokenEquation: SearchBarTokenEquation,
+): SearchBarTokenEquation {
+  const freeTextOperators = [
+    "like",
+    "not like",
+    "is in",
+    "is not in",
+    "in the last",
+    "<",
+    "<=",
+  ];
```

Comment:
Again, I think it would be better to define them in a single place, to avoid having errors if one of these values change at some point.